Sources/Widgets/Widgets3D/LineWidget/index.js

```diff
@@ -0,0 +1,165 @@
+      default:
+        handleRepresentation[0] = vtkSphereHandleRepresentation;
+        break;
+    }
+    switch (model.shapeHandle2) {
+      case 'sphere':
+        handleRepresentation[1] = vtkSphereHandleRepresentation;
+        break;
+      case 'cube':
+        handleRepresentation[1] = vtkCubeHandleRepresentation;
+        break;
+      case 'cone':
+        handleRepresentation[1] = vtkConeHandleRepresentation;
+        break;
+      default:
+        handleRepresentation[1] = vtkSphereHandleRepresentation;
+        break;
+    }
+    handleRepresentation[2] = vtkSVGCustomLandmarkRepresentation;
+  }
+
+  model.methodsToLink = [
+    'activeScaleFactor',
+    'activeColor',
+    'useActiveColor',
+    'glyphResolution',
+    'defaultScale',
+  ];
+  model.behavior = widgetBehavior;
+  model.widgetState = stateGenerator();
+  detectHandleShape();
+
+  publicAPI.getRepresentationsForViewType = (viewType) => {
+    switch (viewType) {
+      case ViewTypes.DEFAULT:
+      case ViewTypes.GEOMETRY:
+      case ViewTypes.SLICE:
+      case ViewTypes.VOLUME:
+      default:
+        return [
+          {
+            builder: handleRepresentation[0],
+            labels: ['handle1'],
+            initialValues: { scaleInPixels: true },
+          },
+          {
+            builder: handleRepresentation[1],
+            labels: ['handle2'],
+            initialValues: { scaleInPixels: true },
+          },
+          { builder: handleRepresentation[2], labels: ['text'] },
+          //    {
+          //     builder: vtkLineHandleRepresentation,
+          //    labels: ['handle1', 'handle2', 'moveHandle'],
+          //  initialValues: { scaleInPixels: true },
+          // },
+          {
+            builder: vtkPolyLineRepresentation,
+            labels: ['handle1', 'handle2', 'moveHandle'],
+            initialValues: { scaleInPixels: true },
+          },
+        ];
+    }
+  };
+
+  // --- Public methods -------------------------------------------------------
+
+  publicAPI.getDistance = () => {
+    const handles =
+      model.widgetState.getHandle1List() + model.widgetState.getHandle2List();
+    if (handles.length !== 2) {
+      return 0;
+    }
+    return Math.sqrt(
+      distance2BetweenPoints(handles[0].getOrigin(), handles[1].getOrigin())
+    );
+  };
+
+  // --------------------------------------------------------------------------
+  // initialization
+  // --------------------------------------------------------------------------
+
+  model.widgetState.onBoundsChange((bounds) => {
+    const center = [
+      (bounds[0] + bounds[1]) * 0.5,
+      (bounds[2] + bounds[3]) * 0.5,
+      (bounds[4] + bounds[5]) * 0.5,
+    ];
+    model.widgetState.getMoveHandle().setOrigin(center);
+  });
+
+  // Default manipulator
+  model.manipulator = vtkPlanePointManipulator.newInstance();
+}
+
+// ----------------------------------------------------------------------------
+
+const DEFAULT_VALUES = {
+  // manipulator: null,
```

Comment:
As you add a setGet macro for this attribute, maybe that could be good to initialize it.

Reply:
VTK.js coding style does not advice to initialize a member variable with `null`.